src/api_manager/fetch_metadata.cc

```diff
@@ -118,11 +140,15 @@ void GlobalFetchServiceAccountToken(
                     continuation(Status(Code::INTERNAL, kFailedTokenFetch));
                     return;
                   }
-
-                  if (!token->SetTokenJsonResponse(body)) {
-                    env->LogDebug("Failed to parse token response body");
-                    continuation(Status(Code::INTERNAL, kFailedTokenParse));
-                    return;
+                  if (audience.empty()) {
+                    if (!token->SetTokenJsonResponse(body)) {
+                      env->LogDebug("Failed to parse token response body");
+                      continuation(Status(Code::INTERNAL, kFailedTokenParse));
+                      return;
+                    }
+                  } else {
+                    token->set_access_token(body,
+                                            kInstanceIdentityTokenExpiration);
```

Comment:
so identify token response doesn't have expiration time?  only token?

Reply:
yes. 
manually set kInstanceIdentityTokenExpiration = 3500.

Reply:
Add a //TODO to parse JWT to get expiration time

Reply:
Done.